tools/ingest_csv.py

```diff
@@ -0,0 +1,195 @@
+    else:
+        props = None
+
+    # Creating parrent asset
+    if not asset_names[-2]:
+        raise InvalidCSVException(f"Parent asset not defined in {csv_file} in row {row}")
+    parent_asset = ParentAsset(
+        nodes=nodes_asset.nodes, parent=Asset(name=asset_names[-2], type=asset_types[-2]), props=props
+    )
+
+    # Creating leaf asset - optionally
+    if asset_names[-1]:
+        return LeafAsset(
+            nodes=nodes_asset.nodes,
+            parent=parent_asset.parent,
+            leaf=Asset(name=asset_names[-1], type=asset_types[-1]),
+            props=props,
+        )
+
+    return parent_asset
+
+
+def _create_source_code(
+    source_code_text: str, highlights: str, transformation_display_name: str, custom_lineage_config: CustomLineageConfig
+) -> Optional[SourceCode]:
+    if not source_code_text:
+        return None
+
+    if highlights:
+        source_code_highlights = []
+        for highlight in highlights.split(","):
+            source_code_highlights.append(
+                SourceCodeHighLight(start=highlight.split(":")[0][1:], len=highlight.split(":")[1][:-1])
+            )
+    else:
+        source_code_highlights = None
+
+    return generate_source_code(
+        source_code_text=source_code_text,
+        custom_lineage_config=custom_lineage_config,
+        transformation_display_name=transformation_display_name,
+        highlights=source_code_highlights,
+    )
+
+
+def ingest_csv_files(source_directory: str, custom_lineage_config: CustomLineageConfig) -> None:
+    source_dir = Path(source_directory)
+    unique_asset_types: Set[str] = set()
+
+    # Verify source directory exists
+    if not source_dir.exists():
+        raise InvalidCSVException(
+            f"Could not find {source_dir}, please make sure to provide a correct source directory."
+        )
+
+    # Extract csv files from source directory
+    csv_files = [f for f in source_dir.iterdir() if f.is_file() and f.suffix == ".csv"]
+    if not csv_files:
+        raise InvalidCSVException(
+            f"No csv files found in {source_dir}, please make sure to provide directory with csv files."
+        )
+
+    # Extract the lineage relationships from the csv files
+    lineages = []
+    for csv_file_to_ingest in csv_files:
+        with open(csv_file_to_ingest, "r") as csv_file:
+            csv_reader = csv.reader(
+                csv_file,
+            )
+            headers = next(csv_reader)
+            index_fullname = _validate_header(headers=headers, csv_file=csv_file_to_ingest)
+            unique_asset_types.update(headers[: index_fullname[0]])
+            unique_asset_types.update(headers[index_fullname[0] + 2 : index_fullname[1]])
+            for row in csv_reader:
+                if len(row) != len(headers):
+                    raise InvalidCSVException(
+                        f"Row {row} in file {csv_file} does not contain same amount of entries as the headers"
```

Comment:
Maybe we should put the line number in the error message, could be easier for big files to fix the wrong line .